streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -495,7 +495,7 @@ StandbyTask createTask(final Consumer<byte[], byte[]> consumer,
 
             final ProcessorTopology topology = builder.build(taskId.topicGroupId);
 
-            if (!topology.stateStores().isEmpty()) {
+            if (!topology.stateStores().isEmpty() && !topology.storeToChangelogTopic().isEmpty()) {
```

Comment:
Does this still work for optimized source tables, which read from the input topic instead?

Reply:
Good point. Thinking about this, StandBys for source-KTables might have been broker for a long time already... (maybe since 0.10.0.0???)

Maybe @cadonna can verify? If that is the case, we should split out a separate ticket and PR to fix StandBys for source-KTables independently.

Reply:
Yeah, good point! Added an integration test to verify materialized and optimized source tables.

Reply:
Cannot follow. You test seem to use the PAPI, and the PAPI does not provide the KTable optimization. You would need to use `StreamBuilder#table()` to test the changelog optimization.

Reply:
See `StandbyTaskCreationIntegrationTest` line 128.